service/settings.py

```diff
@@ -198,13 +209,33 @@ def autodetect(self):
 
     def getProxySettings(self):
 
-        if self.getMode() == 0:
+        if self.getMode() == self.PROXY_MODE_NONE:
             return None
-        if self.getMode() == 1:
+        if self.getMode() == self.PROXY_MODE_AUTODETECT:
             return self.autodetect()
-        if self.getMode() == 2:
+        if self.getMode() == self.PROXY_MODE_MANUAL:
             return (self.getAddress(), int(self.getPort()))
 
+    def getProxyAuthDetails(self):
+        if self.getMode() == self.PROXY_MODE_NONE:
+            return None
+        if (self.serviceNetworkSettings["login"] is None) or (self.serviceNetworkSettings["password"] is None):
+            return None
+        # in all other cases, return tuple of (login, password)
+        return (self.serviceNetworkSettings["login"], self.serviceNetworkSettings["password"])
```

Comment:
I used tuple because: 
1) I saw similar code was for proxy address and port return https://github.com/pyfa-org/Pyfa/blob/master/service/settings.py#L206 ```return (self.getAddress(), int(self.getPort()))```
2) I left a comment near the code that is using the result
```self.nAuth = self.settings.getProxyAuthDetails()  # tuple of (login, password)```

Agree with @Ebag333 that dict with keys makes code a bit more readable, will do whatever you like.